ocaml/xapi/xapi_pool.ml

```diff
@@ -2926,7 +2930,10 @@ let remove_repository ~__context ~self ~value =
     Repository.with_reposync_lock @@ fun () ->
     List.iter
       (fun x ->
-         if x = value then Repository.cleanup_pool_repo ~__context ~self:x)
+         if x = value then (
+           Repository.cleanup_pool_repo ~__context ~self:x ;
+           Repository.reset_updates_in_cache ()
```

Comment:
I would have expected `cleanup_pool_repo` to do this reset anyway. Should `set_up_to_date` be doing this reset also? If so then we don't need to expose `reset_updates_in_cache` anymore in the respository.mli file

Reply:
The case of `set_up_to_date` is actually to add new repositories, which indeed requires `reset_updates_in_cache`.

Reply:
Then can we keep `reset_updates_in_cache` as an implementation detail of `repository.ml`? Based on your comment I don't see a reason to expose it

Reply:
I mean i.e. in the function `add_repository`, there is no other functions exposed from `module Repository` to cover the reset bits. Basically only those `Db.Repository.*` functions are used in this function.

Reply:
If all the repository functions need to call reset they can do that themselves like @lippirk said, so here you could call `Repository.cleanup_pool_repo ~__context ~self:x` and that call would internally call the reset function, and similarly for all the other Repository functions.

Reply:
Please bear with me as I still can't get the point:
For example, in this function, I can't see which one could take the `reset` bits on?
```
let add_repository ~__context ~self ~value =
  Xapi_pool_helpers.with_pool_operation
    ~__context
    ~self
    ~doc:"pool.add_repository"
    ~op:`configure_repositories
    @@ fun () ->
    Repository.with_reposync_lock @@ fun () ->
    let existings = Db.Pool.get_repositories ~__context ~self in
    if not (List.mem value existings) then (
      Db.Pool.add_repositories ~__context ~self ~value ;
      Db.Repository.set_hash ~__context ~self:value ~value:"" ;
      Db.Repository.set_up_to_date ~__context ~self:value ~value:false ;
      Repository.reset_updates_in_cache ()
    )
```
@lippirk  suggested `set_up_to_date`, but this is a function in `Db`. I don't think I could overwrite it.

Reply:
> If all the repository functions need to call reset they can do that themselves

Some of the functions that need to call reset are implemented here, like `add_repository`. So is does need the reset to be exposed for those. Then for consistency, I'd make all reset calls explicit in the functions in this module (xapi_pool), as Ming has done.